### ISSUE TITLE
[PEP 541] Spam account detected

### ISSUE THREAD

**Issue by webknjaz** [pypa/pypi-support#536]
**Reasons for the request**

I was looking at https://pypi.org/search/?c=Topic+%3A%3A+System+%3A%3A+Software+Distribution trying to figure out if this classifier fits the needs of my project by example.

And what I noticed is that the first page is literally flooded with `Reserve names for currrent development items until release` projects. 11 entries out of 20 are dummy empty projects. They all turned out to be created by https://pypi.org/user/acue/ and if you look there, it has 563 spam projects like this.

I'm pretty sure it's an example of https://www.python.org/dev/peps/pep-0541/#invalid-projects.

P.S. GitHub user seems to be @ArnoCan.

**Comment by ArnoCan** [comment]
Hello Sviatloislav,
this is actually no spam.

I am working on them and if you look into the names, they make sense as for
example apiref is a (successor of epdoc) and the modules names 'apiref-*'
represent additional syntaxes. It takes a bit of time because I need a
complete platform - here for Python3.5+ AND Python2.7 -  I am going to
continue with support for 2.7.

The reason for the names is simple, I had tried to allocate some projects
on pypi as well as on sourceforge.net, and in case of master projects with
modules on github as well as bitbucket. These names have to be unique, not
so the user projects on github and bitbucket, but in general on PyPI and
Sourc eforge. I made the experience, that I had in the most cases for a bit
sounding names old, outdated and in some cases empty projects since several
years which occupied the names. So I tried to assure my woking names,
because I am writing the documentation in parallel and had to cross
reference by these names in code as well as in documents, which are not in
any case simply changeable by sed/awk.

So please be patient until I have finished the first bunch of consistent
packages and have released them.

Each name is actually a package, so no spam.

WKR
Arno

Am Di., 4. Aug. 2020 um 22:10 Uhr schrieb Sviatoslav Sydorenko <
notifications@github.com>:

**Comment by ArnoCan** [comment]
Hello Sviatloislav,
another remark, I am writing some common frameworks for multiple platforms
which I require for the top-level applications. Thus it is a bit a
recursive adaptation and not a streamlined slim development. So the first
set of packages has to be consistent including their APIs. Thus I am not
publishing this by each package, even though I am already using them, but
in sets of packages.

E.g. a common cross-compiler frontend for data description languages
including JSON, YAML, XML, INI(with several variants), which also is able
to scan arbitrary in-memory Python data structures. With additional
syntaxes such as ASN.1 and application layer SNMP. I am already using
parts, but e.g. had still to adapt for some XML mappings as well as e.g.
syntax specials of YAML. Already published, but not yet updated are
jsondata(part of the 'json*' packages) and yapydata(low-level JSON, XML,
YAML, INI).

So the final set of packages could be published as soon as the updated
internal AST is ready to represent all supported syntaxes. Therefore I have
reserved the names for the virtual major package and the  syntax components.

But anyhow I am going to fill up the package descriptions a bit more
detailed and related to the specific packages.

WKR
Arno


Am Fr., 18. Sept. 2020 um 10:42 Uhr schrieb Arno-Can Üstünsöz <
acue.ch@gmail.com>:

**Comment by jorgecarleitao** [comment]
I agree with @webknjaz here: this is the very definition of name squatting under PEP 541 and IMO you (@arnoCan) should have suspended unless you remove these from pypi: an admin should not have to go through 500 projects to decide whether each and every one of them is squatting or not: the fact that you squatted 500(!!) different names IMO already implies that you are abusing the system.

Please free the projects and re-think your approach: even the largest Python projects in the ecosystem do not perform name squatting like this.

One suggestion: find a name for your project, say `project-x`, and use names such as `project-x-some`, `project-x-other`.

cc @jamadden

**Comment by ArnoCan** [comment]
@webknjaz + @jorgecarleitao  + @jamadden

This is neither spam, nor name squatting.

Same as #659 - e.g. the project 'datafusion' is a project providing a common platform for various data description syntaxes - e.g. JSON, XML, YAML, also INI/CFG/CONF, .properties, with additionals like ASN.1. So this is neither a dummy, nor spam. Due to my development approach which includes thorough documentation I required the names early, because simply using sed for working named does not work in any case and for any cross-refrence in hierarchical package dependencies, neither for development trees.

Another hint for the number of projects.

The projects related to PyPA or setuptools are e.g.:
    setuplib + setuplibcore + setupdocx + setupjavax + setuptestx + setupbashx

These are logically grouped into the virtual packages - similar to e.g. PyWin:
   ictstacks:   which is the overall framework
   packageit: which groups the installer releated packages - here setuptools / distutils

This results in the logical/virtual package hierarchy for installation:
- itstacks -> packageit -> setupbashx
- itstacks -> packageit -> setupdocx
- itstacks -> packageit -> setupjavax
- itstacks -> packageit -> setuplib
- itstacks -> packageit -> setuplibcore
- itstacks -> packageit -> setuptestx

This already makes 8 packages - you can find the most of them already published, but not yet updated - which will be done soon.

So be sure that each package actually exists and supports Python2.7 + Python3.5+ on the platforms Linux, BSD, Unix, OS-X/macOS, and Windows(mostly from W2K on). Additionally OSs like Minix, RisOS(?) and for some OS-alike such as MicroPython.
As you may expect this requires a lot of adaptation, and even more tests(see also platformids and pythonids - available and going to be updated with the first set).

Also to mention the basically required sophisticated document-automation to cope the mass of projects timely - which I had in addition to develop finally by myself(apyref and apyref-*(about 35 syntax packages).

This are some of the reasons for the delayed publishing.

WKR

**Comment by ArnoCan** [comment]
Just one extension
- setupjavax

or
- ictstacks -> packageit -> setupjavax

is required for Jython  - which supports currently Python-2.7 syntax with Java specifics.
All packages support the full scope of PythonIDs - CPython, IPython(specifics), IronPython, Jython, PyPy, and Stackless, where Jython and IronPython currently support Python-2.7 syntax only - on all platforms supported by PlatformIDs.

WKR

**Comment by webknjaz** [comment]
This still counts as name squatting as defined by PEP 541: https://www.python.org/dev/peps/pep-0541/#invalid-projects

**Comment by ArnoCan** [comment]
You should in particular be aware, that this is not a spam-account at all! There are probably too much real projects with more than 100k LOC present by this account. The comprehensive documents are available by link e.g. on <project>.sourceforge.io. With more going to follow step-by-step as planned and already allocated on PyPI.

As far as I see it is not the intention of the PEP-541 to declare a project at work as a squatting project - because anyone could push some pre-alpha into the field which is as useful as none. So I do definitely not agree. 

I recommend to scan the projects with access to the database for actual squatting projects. During the name definition phase I struggled over a number of projects in PyPI which since more than 5-10 years never committed any code at all or have some "few-lines" of dummy code. In particular claiming more attractive names - ( only? ). Having a look into these could be actually helpful. Because it required some effort until I had collected the available names I allocated them in before the upload - of course! This has to be done in case of #659 as I had to do before.

A good and proven example of actual squatting is on github.com named "i4p" - since 2010 never committed a line. I got the written trademark from the patent office 1997 and using this as a brand since than continuously, so own the authorship and though the copyright definitely. 

Give me a short time and I try to find within the next days some of the actual squatting projects in PyPI I failed about to find an appropriate name - just as an help for you. But anyhow, the administrator with access to the site data should be able to find these by a few queries only.

WKR

**Comment by pradyunsg** [comment]
@ArnoCan As it stands, there are many packages under your username (at least the ones that I have inspected) that don't contain any code -- they're empty projects, and don't provide anything beyond declaring the metadata and package name. PEP 541 clearly lists this as an "invalid" project:

> A project published on the Package Index meeting ANY of the following is considered invalid and will be removed from the Index: [snip]
> - project is name squatting (package has no functionality or is empty);

In other words, as far as I can tell, nearly all the packages listed under your username are empty and, thus considered invalid.

> actual squatting is on github.com named "i4p" ([@i4p](https://github.com/i4p)) - since 2010 never committed a line

This is an issue tracker for PyPI. If you want to report name squatting on GitHub/SourceForge/BitBucket, please contact their support and look at their name retention policies. As it stands, these are a lot of namesquats, all of which are subject to removal by PyPI Administrators.

---

If you'd like to report additional namesquatting packages on PyPI, please feel free to file new issues on this tracker. Note that it does not mean that existing reports (such as this one) would be ignored.

**Comment by ArnoCan** [comment]
@pradyunsg 

I do not agree to your interpretation, the projects have functionalities. As I listed before, If zou have to write documentation and are developing cross-dependent packages you need a lot of actual references, which should not change.
Though you need reliable names - which require the preparation of the upload, though the creation of the package. A pretty natural and causal process.

1. The PEP-541 says:
- "(package has **no functionality or is empty**)"
the projects have functionality so do not match the cause.

2. The PEP-541 says:
- "the project **violates copyright**, trademarks, patents, or licenses;
The project-name-trolling would meet this cause, because I am as causal the pre-copyright owner.
Thus the PEP-541 has at least this issue and prohibits in case of actual projects as this is the case it's aplication.

And to say it once again - honestly I am a bit astonished - the PEP-541 is not meant to troll project names with by projects which decide to do so a week before!
The first set of relevant packages will be published today on a number of repositories, so the proof is obvious.

But anyhow - this issue has to be closed in any case,  because it is simply not a spam account. So I definitely reject the issue #541 as baseless - simply a mistake.

WKR

**Comment by webknjaz** [comment]
> 1. The PEP-541 says:
> 
> * "(package has **no functionality or is empty**)"
>   the projects have functionality so do not match the cause.

The PEP operates the terms in the context of PyPI. It's not an abstract project, it's what is actually published on PyPI (the dedicated web-page with a name for the package is its project). In the context of PyPI it doesn't matter if the source exists elsewhere, be it on some code hosting site or your local laptop. If `pip install` only downloads garbage to the end-user's computer, it counts as empty and is name squatting.

**Comment by ewjoachim** [comment]
>> "(package has no functionality or is empty)"
> the projects have functionality so does not match the cause.

I think you misread the "or" in that sentence.

**Comment by jorgecarleitao** [comment]
Quite honestly, @ArnoCan , I think that you are trolling and you know it:

* This is not a new project with a week old: this behavior has been going on for at least 10 months:
    * #198 - Feb 13th - 10 months ago
    * #397 - May 18th - 7 months ago
    * #536 - Aug 4th -  2 months ago
    * #659 - 22 days ago
* you have 500 squatted names which you have done nothing about for the past 10 months
* When asked to rectify your behavior, you instead double-down and started presenting deceiving arguments

You are going against Pypi's ToC, as well as not taking action when asked to rectify the situation. IMO your last comment is circumstantial evidence of malicious intent: deceiving the administrators of your intent (as evidenced by your past 10 months of actions).

Please remove the packages and free them up for other people to use. You are negatively impacting the Python community as a whole.

**Comment by webknjaz** [comment]
> And to say it once again - honestly I am a bit astonished - the PEP-541 is not meant to troll project names by projects which decide to do so a week before!
> 
> * https://test.pypi.org/project/datafusion/#history (11 Oct. 2020)

I don't see a point here. TestPyPI is a separate staging deployment that is meant to be used for experiments, in particular by folks who are new to Python packaging.

**Comment by ArnoCan** [comment]
Astonishing.

Please close the issue and stop claiming my account as a spam account. Which is simply ridiculous.

So I proceed with the deployment as planned today including PyPI and other repositories.

WKR

**Comment by webknjaz** [comment]
Definitely a troll

**Comment by ArnoCan** [comment]
@webknjaz
Dear Sviatoslav,
what is your role/task here?

WKR

**Comment by webknjaz** [comment]
Does calling out for unfair use of PyPI that violates its ToC? I filled a report according to PEP 541 (that knowledgeable folks from the community totally agree with, including PyPI admins). The reason is that nobody wants to see spam in the search. FWIW I'll leave it to the admins to take a reasonable action.

**Comment by ArnoCan** [comment]
@ewjoachim
definitely not

@webknjaz
simply astonishing

@webknjaz last comment
See bitbucket for a better solution concerning the search results :*), a limit of the query and search design could not interpreted  by falsely incriminating some one as spammer and trying to troll it's assets. That is actually ridiculous. Definitely not appropriate for an open source foundation against it's comitters.

@jorgecarleitao
This is definitely wrong and you know that, this looks to me like inverted-proactive argumentation.
Nice trial.
You should first quote actual dummy projects which live here for 5+ years, there are more than enough.
The widespread deployment is going on.
This is an interesting story at least.

See also #703 + #702.

So I request honestly the withdraw of this issues (#536 + #659).

WKR

**Comment by ArnoCan** [comment]
@webknjaz
Dear Sviatoslav,
still interested in your role/task here?

WKR

**Comment by webknjaz** [comment]
@ArnoCan Python and PyPA communities pride themselves as open and considerate which implies a certain behavior to be upheld. Your behavior comes close to violating the PyPA and PSF Code Of Conduct. I'd even argue that you crossed the line here. Many folks pointed out, in a clear manner, what exactly you violate with regard to PEP 541. Treating others with such a disrespect and personal attacks must not be happening in the community.

I think it's time to let @di / @gaborbernat / @ewdurbin resolve this matter.

**Comment by ArnoCan** [comment]
Falsely namecalling (acue / @ArnoCan) as a spammer by Sviatoslav Sydorenko #705

WKR

**Comment by pradyunsg** [comment]
I've gone ahead and locked this issue for now. Let's not escalate this further at this time. I'll suggest that everyone take a step away from this for a little while.

I'll unlock this in a week.

**Comment by pradyunsg** [comment]
Since locking issues removes cross-linking on GitHub, adding context here:

@ArnoCan has been filing multiple issues in PyPA issue trackers since this was locked. The initial bunch of issues were closed and they were advised to keep the discussion in #702. Despite being told to not file additional issues, they've continued to do so, with each issue being a different title/request/demand. As far as I can tell, the relevant issues are #702, #703, #704, #705, #706, #707, #708, #709, https://github.com/pypa/.github/issues/2. There might be more on other issue trackers, that I haven't received a notification for.

At this time, as per https://github.com/pypa/pypi-support/issues/709#issuecomment-720358380, the user is blocked from interacting with projects and issue trackers under the @pypa organisation, since they've been spamming various PyPA issue trackers and have refused to stop creating new issues despite with multiple individuals telling them to stop doing so.
